pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SingleFileIndexDirectory.java

```diff
@@ -385,10 +385,12 @@ public void removeIndex(String columnName, ColumnIndexType indexType) {
 
   @Override
   public Set<String> getColumnsWithIndex(ColumnIndexType type) {
+    // TEXT_INDEX is not tracked via _columnEntries, so need to call
```

Comment:
Specialize the handling for `TEXT` and use the current logic for other index types for efficiency